memory_ttl.go

```diff
@@ -41,23 +44,48 @@ func NewMemoryWithTTL(ttl time.Duration) *MemoryTTL {
 
 // StartGC starts the garbage collection process in a go routine
 func (r *MemoryTTL) StartGC(gcInterval time.Duration) {
-	r.gcInterval = gcInterval
+	if gcInterval <= 0 {
+		return
+	}
+
+	r.gcTicker = time.NewTicker(gcInterval)
+	r.done = make(chan struct{})
+
 	go func() {
-		for _ = range time.Tick(gcInterval) {
-			for key := range r.cache.items {
-				if !r.isValid(key) {
-					r.Delete(key)
+		for {
+			select {
+			case <-r.gcTicker.C:
+				now := time.Now()
+
+				r.Lock()
```

Comment:
Thanks @rjeczalik for implementing this! I was thinking on a more robust solution where we have a deadline for cleaning up items. In the case where we have many many items waiting in the cache whether they are are valid or invalid - does not matter - we may block all the reads from cache for a long time. I am not proposing to implement a "garbage collection" algorithm, since Go's map access is randomized enough and we have checks on "Get"  operations, we can operate on the cache list for a configurable amount of time (say 20ms) then return from the operation. This approach also may lead to issues where excessive usage of memory. We can send a signal in the cases we hit the deadline and the consumers can configure their "deadlineTimeout" value - if they need to.

wdyt?